docs/core/runtime-config/envvars.md

```diff
@@ -0,0 +1,109 @@
+|`COMPlus_FinalizeOnShutdown`|0 or 1|When enabled, on shutdown, it blocks all user threads and calls finalizers for all finalizable objects, including live objects.|
+
+## Assembly loading
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|COMPlus_designerNamespaceResolution|0 or 1|Enables (1) or disables (0) the <xref:System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMetadata.DesignerNamespaceResolve?displayProperty=nameWithType> event for WinRT types.|
+
+## Debugging
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPlus_DbgPackShimPath`|*string_path*|Defines the CoreCLR path to dbgshim.dll. Support for this environment variable may be removed in the future.|
+|`COMPlus_DbgRedirectApplication`|*string_path*|Specifies the auxiliary debugger application to launch.|
+|`COMPlus_DbgRedirectAttachCmd`|*string_params*|Specifies command parameters for attaching the auxiliary debugger.|
+|`COMPlus_DbgRedirectCommonCmd`|*string_format*|Specifies a command line format string for the auxiliary debugger.|
+|`COMPlus_DbgRedirectCreateCmd`|*string_params*|Specifies command parameters when creating the auxiliary debugger.|
+|`COMPLUS_EnableDiagnostics`|0 or 1|Enables (1) or disables (0) debugger and profiler diagnostics.|
+
+## Garbage collection
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPLUS_gcAllowVeryLargeObjects`|0 or 1|Enables (1, the default value) or Disables (0) support on 64-bit platforms for arrays that are greater than 2 gigabytes (GB) in total size.|
+|`COMPLUS_GCCpuGroup`|0 or 1|Enables (1) or disables (0, the default value) support for CPU groups by the garbage collector if server garbage collection is enabled.|
+|`COMPLUS_GCLatencyLevel`|see <xref:System.Runtime.GCLatencyMode>|Adjusts the intrusiveness of the garbage collector. The default value is 1 (<xref:System.Runtime.GCLatencyLevel.Interactive?displayProperty=nameWithType>). For more information, see [Latency modes](../../standard/garbage-collection/latency.md).
+|`COMPLUS_GCName`|*string_path*|Specifies a path to the library containing the GC that the runtime intends to load. For more information, see [Standalone GC Loader Design](https://github.com/dotnet/coreclr/blob/4c7c066d0bacdb86a2311333de1ca73d94ae5280/Documentation/design-docs/standalone-gc-loading.md).|
+|`COMPLUS_GCPollType`|0 or 1|Enables (1) or disables (0, the default value) [`JIT_PollGC`](https://github.com/dotnet/coreclr/blob/deb00ad58acf627763b6c0a7833fa789e3bb1cd0/src/vm/jithelpers.cpp#L6331-L6536) calls to be inserted into user code.|
+|`COMPLUS_GCStress`|0 or 1|If on (1), triggers garbage collections at regular intervals. The default is off (0).|
+|`COMPLUS_GCStressStart`|*n*|Start GCStress after *n* stress GCs have been attempted. The default value is 0.|
+|`COMPLUS_gcTrimCommitOnLowMemory`|0 or 1|When enabled (1), aggressively trims the committed memory for the ephemeral segment to keep as little memory committed as possible. The default is off (0).|
```

Comment:
I know we already have this documented (someone asked to document without consulting the GC team); this should be not public to begin with. this is not handled in core. I should just remove this from the code.